src/device/usbd.c

```diff
@@ -1036,4 +1041,60 @@ bool usbd_edpt_stalled(uint8_t rhport, uint8_t ep_addr)
   return _usbd_dev.ep_status[epnum][dir].stalled;
 }
 
+/**
+ * Remove queued xfer complete messages from event queue,
+ * for a particular ep.
+ */
+static void usbd_abort_transfers(uint8_t rhport, uint8_t ep_addr)
+{
+  dcd_event_t ev_sentinal = 
+  {
+    .event_id = DCD_EVENT_COUNT, ///< This is an invalid event ID.
+  };
+  dcd_event_t event;
+  uint8_t const epnum   = tu_edpt_number(ep_addr);
+  uint8_t const ep_dir  = tu_edpt_dir(ep_addr);
+  
+  dcd_int_disable(rhport);
+  // Queue sentinal element
+  TU_ASSERT(osal_queue_send(_usbd_q, &ev_sentinal, true), /**/);
+
+  TU_ASSERT(osal_queue_receive(_usbd_q, &event), /**/);
+
+  while(event.event_id != DCD_EVENT_COUNT)
+  {    
+    if((event.rhport == rhport) && (event.event_id == DCD_EVENT_XFER_COMPLETE)
+      && (event.xfer_complete.ep_addr == ep_addr))
+    {
+      _usbd_dev.ep_status[epnum][ep_dir].busy = false;
+      event.xfer_complete.ep_addr = 0xFF; // Mark transfer as invalid
+    }
+    TU_ASSERT(osal_queue_send(_usbd_q, &event, true), /**/);
+    TU_ASSERT(osal_queue_receive(_usbd_q, &event), /**/);
+  }
+  
+  dcd_int_enable(rhport);
+}
+
+/**
+ * tud_edpt_close will disable an endpoint, and clear all pending transfers
+ * through the particular endpoint.
+ * 
+ * It must be called from the usb task (i.e. from the control request
+ * handler while handling SET_ALTERNATE).
+ */
+void tud_edpt_close(uint8_t rhport, uint8_t ep_addr)
```

Comment:
edpt_close should not be exported to application, it will be handled by the usbd and dcd only.

Reply:
That means it should use the usbd prefix? Or a different header file?

Reply:
yes, it should be usbd_ prefix and in the `usbd_pvt.h` header file for the class usage only.

Reply:
Changed.